cocotb/drivers/amba.py

```diff
@@ -184,6 +184,8 @@ def read(self, address, sync=True):
 
         raise ReturnValue(data)
 
+    def __len__(self):
+        return 2**len(self.bus.ARADDR)
```

Comment:
As mentioned in the original CR - this doesn't make a whole lot of sense to me, and seems like an abuse of `len`

Reply:
Would you mind filing a new issue to find a better solution for that? The PR and the associated discussion have gotten way too long with many unrelated changed in flight.